programs/relayer/build.rs

```diff
@@ -1,6 +1,12 @@
+use std::env;
+use std::path::PathBuf;
+
 fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let out_dir = PathBuf::from(env::var("OUT_DIR").unwrap());
```

Comment:
This means that `OUT_DIR` is required to build the relayer. Should this be optional?

Reply:
hmm, that's what I thought, but tonic build already uses this variable to output the protos. I'm not sure if there's another way to retrieve the output directory of where the protos go. advice welcome here

Reply:
I can make it optional. (Although it built fine on my machine without the out_dir).  I'm not sure I fully understand how this is supposed to work. Where is the file descriptor actually supposed to be? Is it somewhere like `~/.relayer/relayer_descriptor.bin`?

Because as far as I can tell, all you are doing in `build.rs` is to set a path for it. Isn't it a bit weird that this would depend on the builder's env?

Reply:
it's supposed to go into the same folder as the autogenerated code for the relayer proto (`target/release/build/<build_id>/out`). https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates

cargo automatically sets this variable for builds

Reply:
All good.